src/workers/postTranslated.ts

```diff
@@ -0,0 +1,30 @@
+import { TypedWorker } from './worker';
+import { Post } from '../entity';
+import { validLanguages } from '../types';
+import { logger } from '../logger';
+
+export const postTranslated: TypedWorker<'kvasir.post-translated'> = {
+  subscription: 'api.post-translated',
+  handler: async (message, con) => {
+    const { id, translations, language } = message.data;
+
+    if (!validLanguages.includes(language)) {
+      logger.error({ id, language }, '[postTranslated]: Invalid language');
+      return;
+    }
+
+    const post = await con.getRepository(Post).findOneByOrFail({ id });
+
+    const existingTranslation = post.translation?.[language];
+    post.translation[language] = {
+      ...existingTranslation,
+      ...translations,
+    };
```

Comment:
should we not be able to use json `newFlags || flags` to merge on psql side?

Reply:
Yeah, guess that's possible too